### ISSUE TITLE
Strange behaviour of put function 

### ISSUE THREAD

**Issue by jfcap** [janet-lang/janet#24]
What happens here ?

````
Janet 0.3.0-38a7e4f  Copyright (C) 2017-2019 Calvin Rose

janet:0:> (def t (put @{} :a 1))
@{:source-map ("repl" 1 22) :value <cycle 0>}

janet:23:> (put (put @{} :a 1) :b 2) 
error in repl: expected array|table|buffer, got :b
  in _thunk [repl] at (24:48) (tailcall)
````
same problem with array 

````
janet:50:> (def a (put @[] 0 1))
@{:source-map ("repl" 51 71) :value <cycle 0>}
````

**Comment by bakpakin** [comment]
Seems to be a compiler bug. The source-map stuff is generated by the compiler to put into the _env table,  but it should obviously not be returned.

**Comment by bakpakin** [comment]
Found the problem in specials.c, seems that the inliner for the `put` function had an issue. An internal register was used after being freed. Fixed in f284776490ff4c8f5ad5faa1822e5f4b70961b87.

**Comment by jfcap** [comment]
Thanks !
